src/Adapter/Module/Module.php

```diff
@@ -213,7 +213,12 @@ public function onInstall()
         // "Notice: Use of undefined constant _PS_INSTALL_LANGS_PATH_ - assumed '_PS_INSTALL_LANGS_PATH_'"
         LegacyModule::updateTranslationsAfterInstall(false);
 
-        return $this->instance->install();
+        $result = $this->instance->install();
+        if (1 === $result || 0 === $result) {
+            $result = (bool)$result;
+        }
+
+        return $result;
```

Comment:
Actually, casting as a boolean could be a more suitable change:

```php
return (bool) $this->instance->install();
```

This change should be taken in account in the other methods as well, with adapted PHPDoc. This could help us removing additional checks in the controllers.

Reply:
Check this: src/PrestaShopBundle/Controller/Admin/ModuleController.php:303

Reply:
Switching the different checks on the ModuleController is a good idea. On the opposite, I dislike the way we do something only for 2 specific cases. If we want to teach a module developer to return a boolean, we should not try to intercept and fix an integer value.

Reply:
Yes, but do you want to break a lot of modules that returns the integer on install (may be also modules developed by PrestaShop)? ;)

Reply:
Well these modules are already broken, as they don't pass the 3 existing strict checks `true` / `false` / `null`.

And if one of our native modules (= available on GitHub) if not returning a boolean, let me know as we should fix it.